script.js

```diff
@@ -223,4 +223,21 @@ form.addEventListener('submit', (postIt) => {
   if (!formValidate()) {
     postIt.preventDefault();
   }
-});
\ No newline at end of file
+});
+
+// local storage start here
+
+form.addEventListener('input', () => {
+  const jsonForm = {
+    name: document.querySelector('.form-control1').value,
+    email: document.querySelector('.form-control2').value,
+    textArea: document.querySelector('.form-control3').value
+  };
+
+  localStorage.setItem('contactWay', JSON.stringify(jsonForm));
+});
+
+const nonJsonForm = JSON.parse(localStorage.getItem('contactWay'));
+document.querySelector('.form-control1').value = nonJsonForm.name;
+document.querySelector('.form-control2').value = nonJsonForm.email;
+document.querySelector('.form-control3').value = nonJsonForm.textArea;
```

Comment:
- [ ] As part of the requirements you need to have an if check that checks if the `nonJsonForm` has data. otherwise you get a JavaScript error when you the user does not initial have any localStorage

![image](https://user-images.githubusercontent.com/18730106/152656459-ce632a4f-6ed9-479e-ae05-ee362783df01.png)

something like this would be nice 👇
```suggestion
const nonJsonForm = JSON.parse(localStorage.getItem('contactWay'));
if(nonJsonForm) {
  document.querySelector('.form-control1').value = nonJsonForm.name;
  document.querySelector('.form-control2').value = nonJsonForm.email;
  document.querySelector('.form-control3').value = nonJsonForm.textArea;
}
```

Reply:
Thanks ,I have added the if checker